scripts/admin_redistribute_routers.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python
+"""
+This script will take all users without a 'router_name' credential in Identity and assign it to them.
+Distribution will occur evenly across the list in `provider.get_routers()`, and take into account previously set router_names.
+"""
+import argparse
+import django; django.setup()
+from core.models import Provider, Identity
+
+
+def main(args):
+    provider_id = args.provider
+    redistribute = args.redistribute
+    for provider in Provider.objects.filter(id=provider_id):
+        if redistribute:
+            needs_router = provider.identity_set.all()
+        else:
+            needs_router = provider.missing_routers()
+
+        router_map = provider.get_router_distribution(redistribute=redistribute)  # Print 'before'
+        if redistribute:
+            router_map = {key: 0 for key in router_map.keys()}
```

Comment:
@lenards here is the router mapping that i refer to in the comment-chain above...